src/main/java/com/github/tomakehurst/wiremock/matching/ContentPatternDeserialiser.java

```diff
@@ -16,30 +16,79 @@
 package com.github.tomakehurst.wiremock.matching;
 
 import com.fasterxml.jackson.core.JsonParser;
-import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.BeanProperty;
 import com.fasterxml.jackson.databind.DeserializationContext;
 import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonMappingException;
 import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.ContextualDeserializer;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
 import java.io.IOException;
 import java.util.*;
 
-public class ContentPatternDeserialiser extends JsonDeserializer<ContentPattern<?>> {
+public class ContentPatternDeserialiser extends StdDeserializer<ContentPattern<?>>
+    implements ContextualDeserializer {
+
+  private static final Map<String, Class<? extends ContentPattern<?>>> PATTERNS =
+      new HashMap<>(
+          Map.ofEntries(
+              Map.entry("binaryEqualTo", BinaryEqualToPattern.class),
+              Map.entry("equalTo", EqualToPattern.class),
+              Map.entry("equalToJson", EqualToJsonPattern.class),
+              Map.entry("matchesJsonPath", MatchesJsonPathPattern.class),
+              Map.entry("matchesJsonSchema", MatchesJsonSchemaPattern.class),
+              Map.entry("equalToXml", EqualToXmlPattern.class),
+              Map.entry("matchesXPath", MatchesXPathPattern.class),
+              Map.entry("contains", ContainsPattern.class),
+              Map.entry("not", NotPattern.class),
+              Map.entry("doesNotContain", NegativeContainsPattern.class),
+              Map.entry("matches", RegexPattern.class),
+              Map.entry("doesNotMatch", NegativeRegexPattern.class),
+              Map.entry("before", BeforeDateTimePattern.class),
+              Map.entry("after", AfterDateTimePattern.class),
+              Map.entry("equalToDateTime", EqualToDateTimePattern.class),
+              Map.entry("anything", AnythingPattern.class),
+              Map.entry("absent", AbsentPattern.class),
+              Map.entry("and", LogicalAnd.class),
+              Map.entry("or", LogicalOr.class),
+              Map.entry("matchesPathTemplate", PathTemplatePattern.class)));
+
+  public ContentPatternDeserialiser() {
+    super(ContentPattern.class);
+  }
+
+  private ContentPatternDeserialiser(Class<?> vc) {
+    super(vc);
+  }
+
+  public static void registerPatterns(
+      Map<String, Class<? extends ContentPattern<?>>> contentPatterns) {
+    contentPatterns.forEach(PATTERNS::putIfAbsent);
+  }
```

Comment:
Looking for recommendations for `Extensions` to supply these in a different way. The only alternative I was able to come up with was to use a `ServiceLoader` here to go get them, but that would most likely require an additional `META-INF/services` file specifically for `ContentPatternExtension` and they wouldn't be able to be supplied from `ExtensionFactory`.

Reply:
My concern is that if multiple WireMock servers are created and only one registers the `ContentPatternExtension` all will still have access to use it.